Tasks/PerformActionOnMachineGroup/AzureHelper.ps1

```diff
@@ -60,6 +60,43 @@ function Restart-MachineInProvider
 
 function Initialize-AzureHelper
 {
+    Write-Verbose "Initializing azure resource provider" -Verbose
+
+    Import-AzurePowerShellModule
+
     Switch-AzureMode AzureResourceManager
-    Write-Verbose "Switched to AzureResourceManager" -Verbose
+
+    if($machineGroup.ProviderDataList.Count -gt 0)
```

Comment:
We should get into this code only after checking we have Azure Provider and also have valid Provider Data. If there is no provider data - we should not even get into Initialize-AzureHelper.